### PR TITLE
Don't require 2FA for theme developers versions

### PR DESCRIPTION
**Pull request by diox**
This is achieved by overriding `theme_specific` in `_submit_upload()` if an add-on is passed (or if we're using the theme wizard). That variable is passed down to the upload flow (and checked against the xpi being uploaded later) to bypass the 2FA requirement.

### Context

We enforce 2FA at submission time if the `2fa-enforcement-for-developers-and-special-users` flag is active for the user, but we want that for extensions (and langpacks/dicts, technically), not themes. The bug was that we accidentally required it for all new versions of existing add-ons, even themes.

### Testing

Automated tests should cover this but if you want to try locally you can follow the STR in the issue. You can use the theme wizard to create the initial theme, and then take the zip and bump the version number. 

Note that you need to create the waffle flag (not a switch! it can be potentially active for a limited set of users) with: `manage.py waffle_flag 2fa-enforcement-for-developers-and-special-users --everyone`. Then, when logging in with the fake auth, check or uncheck the "two-factor authentication" checkbox before logging in depending on what you want to test.

Fixes https://github.com/mozilla/addons/issues/1956

### CODE REVIEW

File: src/olympia/files/utils.py [hunk 1]
```diff
@@ -1043,7 +1043,13 @@ def parse_addon(
             msg = gettext(
                 'The type (%s) does not match the type of your add-on on AMO (%s)'
             )
-            raise forms.ValidationError(msg % (parsed['type'], addon.type))
```
Comment (diox) [inline src/olympia/files/utils.py:1046]: This is a drive-by fix for this fairly obscure/rare error message which used to return the constant instead of the "pretty" name. See corresponding test fix. 

Note with 2FA enforcement enabled, it can be difficult to trigger anyway: if your user doesn't have 2FA, and you're trying to upload an extension as a version of an existing theme, you'd get the `Please add two-factor authentication to your account to submit extensions` error message instead, as devhub would detect you're using a theme specific flow to upload an extension. It's still relevant if doing the opposite (uploading a theme as a version of an extension) or when using other types though.

File: src/olympia/constants/base.py [hunk 2]
```diff
@@ -134,7 +134,7 @@
     _ADDON_LPADDON: _('Deprecated Language Pack (Add-on)'),
     _ADDON_PLUGIN: _('Deprecated Plugin'),
     _ADDON_PERSONA: _('Deprecated LWT'),
-    ADDON_STATICTHEME: _('Theme (Static)'),
+    ADDON_STATICTHEME: _('Theme'),
```
Comment (eviljeff) [inline src/olympia/constants/base.py:137]: These are also unrelated changes too, though coming way past time!

File: src/olympia/devhub/views.py [hunk 3]
```diff
@@ -1467,6 +1467,12 @@ def _submit_upload(
     form = forms.NewUploadForm(
         request.POST or None, request.FILES or None, addon=addon, request=request
     )
+    if wizard or (addon and addon.type == amo.ADDON_STATICTHEME):
+        # If using the wizard or submitting a new version of a theme, we can
+        # force theme_specific to be True. If somehow the developer is not
+        # uploading a theme, validation will reject it just like if they had
+        # tried to use the theme submission flow for an entirely new add-on.
+        theme_specific = True
```
Comment (eviljeff) [inline src/olympia/devhub/views.py:1475]: submissions via the wizard already set `theme_specific` kwarg, right?  The patch is just covering the case for existing themes having new versions submitted using the xpi upload flow?

**Review by eviljeff — COMMENTED** [review]
Is the case of using the non-wizard upload flow to submit a _new_ theme covered already?  (Or is that impossible to handle because we don't know at the point we'd need to enforce 2FA that they're submitting a theme xpi)

File: src/olympia/constants/base.py (hunk 2 above)
Comment (diox) [reply src/olympia/constants/base.py:137]: Yes, goes with the drive-by fix mentioned above, mainly to avoid saying `'The type (Extension) does not match the type of your add-on on AMO (Theme (Static))` for instance, which looks weird.

File: src/olympia/devhub/views.py (hunk 3 above)
Comment (diox) [reply src/olympia/devhub/views.py:1475]: Yes. I figured it didn't hurt to make sure `theme_specific` is set even if using the wizard.

**Comment by diox** [comment]
> Is the case of using the non-wizard upload flow to submit a new theme covered already? (Or is that impossible to handle because we don't know at the point we'd need to enforce 2FA that they're submitting a theme xpi)

It's in fact the only new case we're covering in this PR. We know the existing add-on is a theme so we set `theme_specific` in https://github.com/mozilla/addons-server/pull/22169/files/7d0fb5dd599c8697f04648066b5a3a752218ced6#diff-da2e9cbdc57a9d4adb11cb41193b9e44b26cbf1e702d61c3bc0b2775897f1a20. As mentioned in my other comment the wizard was technically already working because the wizard view passes `theme_specific=True` kwarg already.

**Review by eviljeff — APPROVED** [review]
